src/mainboard/clevo/mtl-h/variants/dgpu/gpio.c

```diff
@@ -314,16 +315,21 @@ void mainboard_configure_gpios(void)
 
 	gpio_configure_pads(gpio_table, ARRAY_SIZE(gpio_table));
 
-	/* dGPU power on sequence */
-	mdelay(4);
-	gpio_set(DGPU_PWR_EN, 1);
-	result = wait_ms(200, gpio_get(DGPU_PWRGD) == 1);
-	if (result) {
-		printk(BIOS_INFO, "dGPU powered on\n");
-		gpio_set(DGPU_RST_N, 1);
+	if (dasharo_is_dgpu_enabled()) {
+		/* dGPU power on sequence */
+		mdelay(4);
+		gpio_set(DGPU_PWR_EN, 1);
+		result = wait_ms(200, gpio_get(DGPU_PWRGD) == 1);
+		if (result) {
+			printk(BIOS_INFO, "dGPU powered on\n");
+			gpio_set(DGPU_RST_N, 1);
+		} else {
+			printk(BIOS_ERR, "dGPU failed to power on, turning off\n");
+			gpio_set(DGPU_PWR_EN, 0);
+		}
 	} else {
-		printk(BIOS_ERR, "dGPU failed to power on, turning off\n");
-		gpio_set(DGPU_PWR_EN, 0);
+			gpio_set(DGPU_PWR_EN, 0);
```

Comment:
this is the default state (configured in gpio_configure_pads) so this else block is not necessary